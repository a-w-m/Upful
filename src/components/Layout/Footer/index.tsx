import React from "react"
import {
  FooterContainer,
  FooterNav,
  CopyrightWrapper,
  Address,
  FooterLink,
} from "./styled"
import { useStaticQuery, graphql } from "gatsby"
import { C } from "../../../interfaces"

const Footer: React.FC<{}> = () => {
  
  interface Data {
    allMarkdownRemark: {
      edges: {
        node: {
          frontmatter: {
            title: string
          }
          fields: {
            slug: string
          }
        }
      }[]
    }
    site:C.FooterLinks["site"]
  }

  const footerData: Data = useStaticQuery(footerQuery)

  return (
    <FooterContainer>
      <FooterNav aria-label = "Footer">
        {footerData.site.siteMetadata.menuLinks.categories.map(category => {
          return (
            <FooterLink to={category.slug} key={category.name}>
              {category.name}
            </FooterLink>
          )
        })}
        {footerData.allMarkdownRemark.edges.map(page => {
          return (
            <FooterLink
              to={page.node.fields.slug}
              key={page.node.frontmatter.title}
            >
              {page.node.frontmatter.title.toLowerCase()}
            </FooterLink>
          )
        })}
      </FooterNav>
      <Address>
        <a href={footerData.site.siteMetadata.email}>{footerData.site.siteMetadata.email}</a>
      </Address>
      <CopyrightWrapper>
        Â© Upful {new Date(Date.now()).getFullYear()} Made with full Irations
      </CopyrightWrapper>
    </FooterContainer>
  )
}

export const footerQuery = graphql`
  {
    allMarkdownRemark(filter: { fileAbsolutePath: { regex: "/pages/" } }) {
      edges {
        node {
          fields {
            slug
          }
          frontmatter {
            title
          }
        }
      }
    }
    site {
      siteMetadata {
        email
        menuLinks {
          categories {
            name
            slug
          }
        }
      }
    }
  }
`

export default Footer
