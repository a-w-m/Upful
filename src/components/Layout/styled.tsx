import styled from "styled-components"
import device from "./mediaQuery"

export const Container = styled.div`
  position: relative;
  height: 100vh;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto 1fr auto;
  grid-template-areas:
    "header"
    "main"
    "footer";
  overflow: scroll;
  row-gap: 2rem;
  font-family: var(--main-font);
`

export const MainWrapper = styled.div`
  margin: 0 auto;
  width: 100%;
  padding: 0rem 2rem;
  grid-area: main;
  @media ${device.laptop}{
    padding: 0 4rem;
  }
`

export const Main = styled.main``
