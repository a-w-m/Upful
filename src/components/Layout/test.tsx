import React from "react"
import { render, screen } from "@testing-library/react"
import Layout from "."
import {SnipcartContextProvider} from "test-utils"


describe("Layout", () => {
  const child = <h1>child</h1>
  beforeEach(()=>{
  render(<Layout children= {child}/>, {wrapper: SnipcartContextProvider})
  })

  it("should match snapshot", () => {
    const {container} = render(<Layout children={child} />, {wrapper: SnipcartContextProvider})
    expect(container.firstChild).toMatchSnapshot()

  })

  it("should display header", ()=>{
    const header = screen.getByRole('banner')
    expect(header).toBeInTheDocument()
  })

  it("should display footer", ()=>{
    const header = screen.getByRole('contentinfo')
    expect(header).toBeInTheDocument()
  })

  
  it("should display main", ()=>{
    const header = screen.getByRole('main')
    expect(header).toBeInTheDocument()
  })

  it("should display children", ()=>{
    const heading = screen.getByRole('heading')
    expect(heading).toBeInTheDocument()
  })

  
})
