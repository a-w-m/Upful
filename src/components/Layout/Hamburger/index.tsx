import React, { SetStateAction, Dispatch } from "react"
import { Container, Button } from "./styled"

interface HamburgerProps {
  open: boolean
  setOpen: Dispatch<SetStateAction<boolean>>
}

const Hamburger: React.FC<HamburgerProps> = props => {
  const { open, setOpen } = props

  return (
    <Container open={open} onClick={() => setOpen(prev => !prev)}>
      <Button
        type="button"
        aria-label="open navigation"
        aria-controls="link-list"
        aria-expanded="false"
        open={open}
      >
        <div />
        <div />
        <div />
      </Button>
    </Container>
  )
}

export default Hamburger
