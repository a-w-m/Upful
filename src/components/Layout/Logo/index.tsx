import React from "react"
import { useStaticQuery, Link, graphql } from "gatsby"
import { GatsbyImage } from "gatsby-plugin-image"
import { LogoWrapper, LogoLink } from "./styled"
import { P } from "../../../interfaces"

interface Data {
  file: {
    id: string
    childImageSharp: P.Image["childImageSharp"]
  }
}

const Logo: React.FC<{}> = () => {
  const data: Data = useStaticQuery(graphql`
    {
      file(
        sourceInstanceName: { eq: "images" }
        relativeDirectory: { eq: "logo" }
      ) {
        id
        childImageSharp {
          gatsbyImageData(layout: CONSTRAINED, width: 250)
        }
      }
    }
  `)

  return (
      <LogoLink to="/">
        <GatsbyImage
          image={data.file.childImageSharp.gatsbyImageData}
          alt="logo"
          style={{ borderRadius: "100%" }}
        />
      </LogoLink>  )
}

export default Logo
