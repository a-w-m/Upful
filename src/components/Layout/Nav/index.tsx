import React from "react"
import { Menu, NavLink } from "./styled"
import Logo from "../Logo"
import { graphql, useStaticQuery } from "gatsby"
import { C } from "../../../interfaces"

interface NavProps {
  open: boolean
}

const Nav: React.FC<NavProps> = props => {
  const { open } = props

  const data: C.MenuLinks = useStaticQuery(categoryQuery)

  return (
    <Menu open={open} id = {"menu"} aria-label = "Main">
      <Logo />
      {data.site.siteMetadata.menuLinks.categories.map(category => {
        return (
          <NavLink to={category.slug} key={category.name}>
            {category.name.toUpperCase()}
          </NavLink>
        )
      })}
    </Menu>
  )
}

export const categoryQuery = graphql`
  {
    site {
      siteMetadata {
        menuLinks {
          categories {
            name
            slug
          }
        }
      }
    }
  }
`

export default Nav
