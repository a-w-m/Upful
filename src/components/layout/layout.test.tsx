import React from "react"
import { render, screen } from "@testing-library/react"
import Layout from "./layout"

describe("Layout component", () => {
  test("should match snapshot", () => {
    render(<Layout children="" />)
    const { container } = render(<Layout children="" />)

    expect(container.firstChild).toMatchSnapshot()
  })

  test("should display correct title and date", () => {
    render(<Layout children="" />)

    expect(screen.getByText("StoreFront")).toBeInTheDocument()
    expect(screen.getByText(/2019/)).toBeInTheDocument()
  })
})
