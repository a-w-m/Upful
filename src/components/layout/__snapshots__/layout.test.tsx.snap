// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should match snapshot 1`] = `
<div
  class="sc-crrsfI cKsulN"
>
  <header
    class="sc-dlfnbm edAUkJ"
  >
    <div>
      <button
        aria-controls="link-list"
        aria-expanded="false"
        aria-label="open navigation"
        class="sc-bdfBwQ fwZoWG"
        type="button"
      >
        <div />
        <div />
        <div />
      </button>
      <nav
        class="sc-gsTCUz eahIHO"
      >
        <a
          href="/"
        >
          About Us
        </a>
        <a
          href="/"
        >
          Clothing
        </a>
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/"
        >
          Accessories
        </a>
      </nav>
    </div>
    <div
      class="sc-hKgILt fEkVZL"
    >
      <h1
        class="sc-eCssSg iHXJeJ"
      >
        <a
          class="sc-jSgupP eywEqP"
          href="/"
        >
          StoreFront
        </a>
      </h1>
    </div>
  </header>
  <div
    class="sc-dQppl gYULKE"
  >
    <main
      class="sc-bqyKva kOAjrj"
    />
  </div>
  <footer
    class="sc-gKsewC fIoWJo"
  >
    <nav
      class="sc-iBPRYJ eeUOoZ"
    >
      <h5
        class="sc-kEjbxe iqwogK"
      >
        Categories
      </h5>
      <a
        href="/"
      >
        About Us
      </a>
      <a
        href="/"
      >
        Clothing
      </a>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/"
      >
        Accessories
      </a>
    </nav>
    <div
      class="sc-fubCfw flkmPR"
    >
      <h5
        class="sc-kEjbxe iqwogK"
      >
        Contact
      </h5>
      <address
        class="sc-iqHYGH fZGVri"
      >
        StoreFront
        <br />
        555 Santa Monica Blvd.
        <br />
        Los Angeles, CA 90210
      </address>
      <address
        class="sc-iqHYGH fZGVri"
      >
        <a
          href="mailto:howdy@storefront.com"
        >
          howdy@storefront.com
        </a>
      </address>
    </div>
    <div
      class="sc-pFZIQ bQODGP"
    >
      <h5
        class="sc-kEjbxe iqwogK"
      >
        Terms & Conditions
      </h5>
      <a
        href="/"
      >
        Returns
      </a>
    </div>
    <div
      class="sc-jrAGrp kxXRxJ"
    >
      Â© StoreFront 
      2019
    </div>
  </footer>
</div>
`;
